### PR TITLE
Upgraded the AM version

### PR DESCRIPTION
**Pull request by eyalk007**
- [x] All [tests](https://github.com/jfrog/jfrog-idea-plugin/actions/workflows/test.yml) passed. If this feature is not already covered by the tests, I added new tests.
-----

Upgraded the Analyzer Manager version.
The AM executes all of our security scans, and returns us the vulnerabilities that were found by the scanners.

The upgrade makes it so the used version of AM uses the new JAS engines, CA scanners-as-data, and new SAST scanners.

### CODE REVIEW

**Review by hadarshjfrog — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -4,7 +4,7 @@ import java.net.http.HttpResponse
 import java.nio.file.Paths
 
 plugins {
-    id "org.jetbrains.intellij" version "1.16.0"
+    id "org.jetbrains.intellij" version "1.17.0"
```
Comment (eyalk007) [inline build.gradle:7]: this is an experiment for something, will remove before merge

File: src/main/java/com/jfrog/ide/idea/inspections/JFrogSecurityWarning.java [hunk 2]
```diff
@@ -62,7 +62,7 @@ public JFrogSecurityWarning(SarifResult result, SourceCodeScanType reporter) {
                 result.getRuleId(),
                 getFirstRegion(result).getSnippet().getText(),
                 reporter,
-                !result.getKind().equals("pass"),
+                (!result.getKind().equals("pass") && (rule.getRuleProperties().map(properties -> properties.getApplicability().equals("applicable")).orElse(true))),
```
Comment (eyalk007) [inline src/main/java/com/jfrog/ide/idea/inspections/JFrogSecurityWarning.java:65]: On one side, this is very long
problem is this is the super, meaning I cant use a private func and call it. 
I can put a defualt value and edit, but that is gross

File: src/main/java/com/jfrog/ide/idea/scan/data/Rule.java [hunk 3]
```diff
@@ -15,6 +16,9 @@ public class Rule {
     @JsonProperty("fullDescription")
     private Message fullDescription;
 
+    @JsonProperty("properties")
+    private RuleProperties properties;
```
Comment (eyalk007) [inline src/main/java/com/jfrog/ide/idea/scan/data/Rule.java:20]: not all rules have properties. should the field itself be optional

File: src/test/resources/sourceCode/applicable_kind_pass_output.sarif [hunk 4]
```diff
@@ -0,0 +1,140 @@
+{
```
Comment (eyalk007) [inline src/test/resources/sourceCode/applicable_kind_pass_output.sarif:1]: just please check i didnt mess things up. the test passes

File: build.gradle (hunk 1 above)
Comment (hadarshjfrog) [reply build.gradle:7]: reminder to remove

File: src/main/java/com/jfrog/ide/idea/inspections/JFrogSecurityWarning.java (hunk 2 above)
Comment (hadarshjfrog) [reply src/main/java/com/jfrog/ide/idea/inspections/JFrogSecurityWarning.java:65]: why not creating a value before it that'll calculate it?
having a map, and getting into and a property of a property of a.... 
isn't a best-practice..

And while you're at it - do the same for line 61

File: src/main/java/com/jfrog/ide/idea/scan/data/Driver.java [hunk 5]
```diff
@@ -48,4 +48,12 @@ public boolean equals(Object other) {
         return (Objects.equals(this.name, rhs.name) && (CollectionUtils.isEqualCollection(this.rules, rhs.rules)));
     }
 
+    public Rule getRuleById(String ruleId) {
```
Comment (hadarshjfrog) [inline src/main/java/com/jfrog/ide/idea/scan/data/Driver.java:51]: please add a test to it, and embed a test for it in: `parseOutputSarif` - where this error is the expected result

File: src/main/java/com/jfrog/ide/idea/scan/data/Rule.java (hunk 3 above)
Comment (hadarshjfrog) [reply src/main/java/com/jfrog/ide/idea/scan/data/Rule.java:20]: is that a question? yes - it should be optional...

File: src/main/java/com/jfrog/ide/idea/inspections/JFrogSecurityWarning.java (hunk 2 above)
Comment (eyalk007) [reply src/main/java/com/jfrog/ide/idea/inspections/JFrogSecurityWarning.java:65]: um okay.
I still dont love it . but its an okay alternative

File: src/test/resources/sourceCode/applicable_kind_pass_output.sarif [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+              "fullDescription": {
+                "text": "The scanner checks whether the vulnerable function `pem.Decode` is called.",
+                "markdown": "The scanner checks whether the vulnerable function `pem.Decode` is called."
+              },
+              "shortDescription": {
+                "text": "Scanner for CVE-2020-28502"
+              }
+            },
+            {
+              "id": "CVE-2022-25978",
```
Comment (hadarshjfrog) [inline src/test/resources/sourceCode/applicable_kind_pass_output.sarif:23]:
```suggestion
              "id": "applic_CVE-2022-25978",
```

File: src/test/resources/sourceCode/applicable_kind_pass_output.sarif [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+              }
+            },
+            {
+              "id": "applic_CVE-2022-29019",
+              "fullDescription": {
+                "text": "The scanner checks whether the vulnerable function `org.apache.xmlbeans.XmlObject.Factory.parse` is called or an interface that extends `org.apache.xmlbeans.XmlObject` is used.",
+                "markdown": "The scanner checks whether the vulnerable function `org.apache.xmlbeans.XmlObject.Factory.parse` is called or an interface that extends `org.apache.xmlbeans.XmlObject` is used."
+              },
+              "shortDescription": {
+                "text": "Scanner for CVE-2020-5310"
```
Comment (hadarshjfrog) [inline src/test/resources/sourceCode/applicable_kind_pass_output.sarif:53]: Please align the CVE numbers with the ID

File: src/test/resources/sourceCode/applicable_kind_pass_output.sarif [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+            "text": "The scanner checks whether the vulnerable function `ansi-regex` is called."
+          },
+          "kind": "pass",
+          "ruleId": "applic_CVE-2021-25878"
+        },
+        {
+          "message": {
+            "text": "The scanner checks whether the vulnerable function `ansi-regex` is called."
+          },
+          "kind": "pass",
```
Comment (hadarshjfrog) [inline src/test/resources/sourceCode/applicable_kind_pass_output.sarif:132]: add one "kind": "fail".
Also - support undetermined and not covered

**Review by hadarshjfrog — CHANGES_REQUESTED** [review]
Please add more information on the PR description - use the full AM name. also explain what is does (JAS Scanners enablement etc...)
Explain what capabilities it brings (new JAS engines, CA scanners-as-data, new SAST scanners)

File: src/main/java/com/jfrog/ide/idea/inspections/JFrogSecurityWarning.java (hunk 2 above)
Comment (eyalk007) [reply src/main/java/com/jfrog/ide/idea/inspections/JFrogSecurityWarning.java:65]: i just made them static instead

File: src/main/java/com/jfrog/ide/idea/scan/data/Driver.java (hunk 5 above)
Comment (eyalk007) [reply src/main/java/com/jfrog/ide/idea/scan/data/Driver.java:51]: i am unsure where to put the test honestly

**Review by hadarshjfrog — APPROVED** [review]

File: src/main/java/com/jfrog/ide/idea/scan/data/Driver.java (hunk 5 above)
Comment (eyalk007) [reply src/main/java/com/jfrog/ide/idea/scan/data/Driver.java]: check again. was added to sarif.

**Review by hadarshjfrog — APPROVED** [review]

**Pull request merged**